docs/api.rst

```diff
@@ -80,26 +79,29 @@ Becomes using HPy API:
 Calling any HPy function on a closed handle is an error. Calling
 ``HPy_Close()`` on the same handle twice is an error. Forgetting to call
 ``HPy_Close()`` on a handle results in a memory leak. When running in
-:ref:`debug-mode:debug mode`, HPy actively checks that you that you don't close a handle
-twice and that you don't forget to close any.
+:ref:`debug-mode:debug mode`, HPy actively checks that you don't
+close a handle twice and that you don't forget to close any. This is
+possible because handles are associated with operations on objects,
+meaning that if a handle leaks, it is possible to identify exactly
+the operation which produced it.
 
 
 .. note::
-  The debug mode is a good example of how powerful it is to decouple the
-  lifetime of handles and the lifetime of an objects.  If you find a memory
-  leak on CPython, you know that you are missing a ``Py_DECREF`` somewhere but
-  the only way to find the corresponding ``Py_INCREF`` is to manually and
-  carefully study the source code.  On the other hand, if you forget to call
-  ``HPy_Close()``, the HPy debug mode is able to tell the precise code
-  location which created the unclosed handle.  Similarly, if you try to
-  operate on a closed handle, it will tell you the precise code locations
-  which created and closed it.
-
-
-The other important difference is that Python/C guarantees that multiple
-references to the same object results in the very same ``PyObject *`` pointer.
-Thus, it is possible to compare C pointers by equality to check whether they
-point to the same object::
+  Debug mode is a good example of how powerful it is to decouple the
+  identity and therefore the lifetime of handles and those of objects.
+  If you find a memory leak on CPython, you know that you are missing a
+  ``Py_DECREF`` somewhere but the only way to find the corresponding
+  ``Py_INCREF`` is to manually and carefully study the source code.
+  On the other hand, if you forget to call ``HPy_Close()``, debug mode
+  is able to identify the precise code location which created the unclosed
+  handle. Similarly, if you try to operate on a closed handle, it will
+  identify the precise code locations which created and closed it.
+
+
+Remember that Python/C guarantees that multiple references to the same
+object results in the very same ``PyObject *`` pointer. Thus, it is
+possible to compare the pointer addresses to check whether they refer
+to the same object::
```

Comment:
I think the reason that tracking handle usage on HPy is easier is much simpler:

In HPy each handle is created (dupped) and closed or returned exactly once within one call to a C extension function.

In the old C API, references may be incremented and decremented across all of the running code over the lifetime of the entire Python interpreter.